### PR TITLE
Set billing address in first checkout step

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because... it moves billing address form and selector from 3rd payment step to 1st address step in checkout page.

#### 1st step:

127 0 0 1_3000_checkout_payment
127 0 0 1_3000_checkout_payment (1)

#### 2nd step:

127 0 0 1_3000_checkout_payment (2)

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] Changes are mentioned in the changelog.

### Test Environment Config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [CU-6yn1rv Move billing address form to 1st step](https://app.clickup.com/t/6yn1rv)

File: src/@next/pages/CheckoutPage/subpages/CheckoutAddressSubpage.tsx [hunk 1]
```diff
@@ -23,58 +23,84 @@ export interface ICheckoutAddressSubpageHandles {
...
 
+  const isShippingRequiredForProducts =
+    items &&
+    items.some(
+      ({ variant }) => variant.product?.productType.isShippingRequired
+    );
   const checkoutShippingAddress = checkout?.shippingAddress
     ? {
         ...checkout?.shippingAddress,
+        email: checkout?.email || user?.email || undefined,
```
Comment (krzysztofwolski) [inline src/@next/pages/CheckoutPage/subpages/CheckoutAddressSubpage.tsx:64]: Does || undefined changes anything?

Comment (orzechdev) [reply src/@next/pages/CheckoutPage/subpages/CheckoutAddressSubpage.tsx]: Could, as `IAddress` interface didn't accept null values. But it shouldn't change anything, so I have fixed it.

File: src/@next/hooks/useCheckoutStepState.ts [hunk 2]
```diff
@@ -16,11 +16,8 @@ export const useCheckoutStepState = (
     );
 
   const getStep = () => {
```
[the remark this answers is not in the document]
Comment (orzechdev) [reply src/@next/hooks/useCheckoutStepState.ts:18]: Your code seems to be cleaner. Changed.

File: cypress/fixtures/example.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
+  "email": "hello@cypress.io",
+  "body": "Fixtures are a great way to mock data for responses to routes"
+}
```
[the remark this answers is not in the document]
Comment (michalina-graczyk) [reply cypress/fixtures/example.json:5]: this one is still not removed

**Pull request merged**